app/controllers/application_controller.rb

```diff
@@ -0,0 +1,30 @@
+class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  before_action :current_user
+  before_action :get_model
+
+  before_action :current_user
+
+  def render_404
+    # DPR: this will actually render a 404 page in production
+    raise ActionController::RoutingError.new('Not Found')
+  end
```

Comment:
Because I copied that part from the MediaRanker Revisited, after the baby helped me refactor my code when I tried to add the option for a suggestion engine, and Kari said y'all would be able to tell it was a recent addition so not to worry about it too much. :)